cartographer_ros/cartographer_ros/playable_bag.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_ROS_PLAYABLE_BAG_H_
+#define CARTOGRAPHER_ROS_PLAYABLE_BAG_H_
+#include <functional>
+#include <queue>
+
+#include "rosbag/bag.h"
+#include "rosbag/view.h"
+#include "tf2_ros/buffer.h"
+
+namespace cartographer_ros {
+
+class PlayableBag {
+ public:
+  // Handles messages as they are about to enter the buffer. Returns a boolean
+  // indicating whether the message should enter the buffer.
+  using FilteringMessageHandler =
```

Comment:
Your comment on this in the code review is essential to understand the code.
Please rename the callback type name and variables that the code becomes self-explaining and/or extend this comment to fully explain the behavior.

Reply:
Okay, updated. I rather documented this in `offline_node.cc` at the [call site](https://github.com/googlecartographer/cartographer_ros/pull/636/files#diff-7153329f5005d51fb75662a602a03f40R214) because that use of the handler is specific to the offline node, not `PlayableBag`. I also renamed the handler to `FilteringEarlyMessageHandler` to hint more at its intended use.